templates/components/request-reset-view.jsx

```diff
@@ -45,8 +47,16 @@ var RequestResetPassword = React.createClass({
       </div>
     );
   },
+  handleTransition: function(page) {
+    this.transitionTo(page.to, '', page.query);
+  },
   handleFormData: function(data) {
-    this.props.submitForm(data);
+    if(!this.refs.userform.state.checkedUsername) {
```

Comment:
Why not try using https://facebook.github.io/react/docs/component-api.html#setstate with a callback to avoid this conflict? Then, you won't need https://github.com/mozilla/id.webmaker.org/pull/277/files#diff-c4fb49d7f3ec54fc1cb603fb9e8b2103R53.

You've tried to set up a lock situation here ( :+1: ), but it causes a race conditions, I think.